CompilerSource/parsing/ast.h

```diff
@@ -509,11 +521,15 @@ class AST {
 
   class Visitor {
     std::ofstream of;
+    bool print_type;
```

Comment:
This will be used with `VisitLambdaExpression`. I prefer not to use it with `VisitFullType` because the default behavior for `VisitFullType` is to print the type, whereas for lambdas, the default is not to print the type.